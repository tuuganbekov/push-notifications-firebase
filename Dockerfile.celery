ARG GOLDENIMAGE
FROM $GOLDENIMAGE
WORKDIR /celery
COPY . .
ENV PATH "/app/.venv/bin:$PATH"
CMD ["celery", "-A", "config", "worker", "-l", "INFO"]