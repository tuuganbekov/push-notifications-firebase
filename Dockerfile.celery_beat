ARG GOLDENIMAGE
FROM $GOLDENIMAGE
WORKDIR /celery_beat
COPY . .
ENV PATH "/app/.venv/bin:$PATH"
CMD ["celery", "-A", "config", "beat", "-l", "INFO"]