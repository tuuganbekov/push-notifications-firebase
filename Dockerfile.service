ARG GOLDENIMAGE
FROM $GOLDENIMAGE
WORKDIR /app
COPY . /app/
ENV PATH "/app/.venv/bin:$PATH"
CMD ["bash", "entrypoint.sh"]
