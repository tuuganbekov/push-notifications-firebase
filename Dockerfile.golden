FROM python:3.10-slim as base

RUN export http_proxy=http://10.230.143.15:3128 && export https_proxy=http://10.230.143.15:3128
ENV PYTHONDONTWRITEBYTECODE=1
ENV PIPENV_VERBOSITY=-1
ENV TZ=Asia/Bishkek

WORKDIR /app

# stage for installing dependencies
FROM base as dependencies
RUN apt-get update \
    && apt-get install -y --no-install-recommends \
    build-essential \
    gcc

RUN python3 -m venv --copies .venv
COPY Pipfile Pipfile.lock ./
RUN . .venv/bin/activate \
&& pip3 install pipenv \
&& pipenv install psycopg2-binary \
&& pipenv install --deploy
